code/datums/components/food/edible.dm

```diff
@@ -462,6 +463,29 @@ Behavior that's still missing from this component that original food items had t
 #undef EAT_TIME_VORACIOUS_MULT
 #undef EAT_TIME_VORACIOUS_FULL_MULT
 
+/datum/component/edible/proc/ThrownBite(obj/item/food, mob/living/carbon/victim, hit_zone, blocked, datum/thrownthing/throwingdatum)
```

Comment:
Hang on a second. This sticks this on literally every edible in the game.

This is not the way to do this. I don't know WHAT the best way to do this might be, but this is definitely not it. And also the culprit for the majority of your runtimes because organs are edible.

Reply:
Thing is, I really don't feel comfortable with food being forcefed by explosions. It might be goofy once or twice but I can see this getting a bit weird given organs are edible and explosions sometimes throw those around.

Reply:
in that case perhaps using such a general signal as the thrown one is a bad idea all round
however the best alternate solution I can think of for this item is unfortunately:
"Make a new element which is attached to things fired from the cannon when they are fired, which fires a new signal on thrown impact and then removes itself" 😅

Reply:
> getting a bit weird given organs are edible and explosions sometimes throw those around.
How is that anything but a harmlessly fun interaction

Reply:
And once it happens once, every subsequent occurrence will get older or weird. Explosions are not that uncommon, and the novelty lasts only so long.

If it was actually a low probability to occur so that most players won't encounter it more than once or twice after hours of player, then this would be just a weird quirky mechanic. But this then becomes increasingly less usable for OP, so...

Reply:
Again it's not weird, old sure, annoying maybe.

But it is definitively not weird.

Reply:
Okay.

Reply:
> > getting a bit weird given organs are edible and explosions sometimes throw those around.
> 
> How is that anything but a harmlessly fun interaction where you might eat an organ if an explosion throws it into your mouth randomly.

There is potential for !fun! since chems can be in the liver i think? someone with death chems that gets blown up (or a realistic scenario you put someone to sleep and blow them up) could possibly result in their liver going into an unfortunate spaceman's mouth